src/components/RTsort/RTsort.tsx

```diff
@@ -17,28 +18,64 @@ interface FormValuesType {
 const rtsort = () => {
     const [loading, setLoading] = useState(false);
     const [output, setOutput] = useState("");
+    const [pid, setPid] = useState("");
 
     let form = useForm({
         initialValues: {
             path: "./",
         },
     });
 
+    // Uses the callback function of runCommandGetPidAndOutput to handle and save data
+    // generated by the executing process into the output state variable.
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); // Update output
+    }, []);
+
+    // Uses the onTermination callback function of runCommandGetPidAndOutput to handle
+    // the termination of that process, resetting state variables, handling the output data,
+    // and informing the user.
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference
+            setPid("");
+
+            // Cancel the Loading Overlay
+            setLoading(false);
+        },
+        [handleProcessData]
+    );
+
     const onSubmit = async (values: FormValuesType) => {
         setLoading(true);
         const args = [`${values.path}`];
 
         const filteredArgs = args.filter((arg) => arg !== "");
 
         // Please note this command should not be cancelled as this will cause the rainbow table to be corrupted
+
         try {
-            const output = await CommandHelper.runCommand("rtsort", filteredArgs);
-            setOutput(output);
+            const result = await CommandHelper.runCommandGetPidAndOutput(
+                "rtsort",
+                filteredArgs,
+                handleProcessData,
+                handleProcessTermination
+            );
+            setPid(result.pid);
+            setOutput(result.output);
         } catch (e: any) {
             setOutput(e);
         }
 
-        setLoading(false);
+        //setLoading(false);
```

Comment:
Why is this commented out?

Reply:
I commented it during testing stage, forgot to remove before pushing,
Updated now.